simplq/src/test/java/me/simplq/controller/advices/AuthenticationFilterTest.java

```diff
@@ -0,0 +1,42 @@
+package me.simplq.controller.advices;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.net.MalformedURLException;
+import me.simplq.exceptions.SQAccessDeniedException;
+import org.junit.jupiter.api.Test;
+
+class AuthenticationFilterTest {
+  private static final String keyUrl = "https://simplq.us.auth0.com/.well-known/jwks.json";
+
+  @Test
+  void denyBadHeaderValues() throws MalformedURLException {
+    var authFilter = new AuthenticationFilter(new LoggedInUserInfo(), keyUrl);
```

Comment:
A unit test should test only one unit of work. Here you test 6. So we can use `@ParameterizedTest`annotation and the method will look like this:
```java
   @ParameterizedTest
   @ValueSource(strings = {null, "", "invalid header"}) //and so on
   void denyBadHeaderValues(String headerValue) throws MalformedURLException { 
        var authFilter = new AuthenticationFilter(new LoggedInUserInfo(), keyUrl);
       assertThrows(SQAccessDeniedException.class, () -> authFilter.authenticate(headerValue));
  }
```